Sources/Build/BuildPlan.swift

```diff
@@ -1091,6 +1091,14 @@ public class BuildPlan {
             buildProduct.additionalFlags += pkgConfig(for: target).libs
         }
 
+        // Mark dylibs safe for application extension if they don't depend on any system module.
+        if buildParameters.triple.isDarwin(), buildProduct.product.type == .library(.dynamic), dependencies.systemModules.isEmpty {
+            // We also ensure the user isn't trying to do anything custom using the -Xlinker style args.
+            if buildParameters.linkerFlags.isEmpty {
```

Comment:
I don't think this is a good idea. The source files will always have been compiled with -fapplication-extension / application-extension so this shouldn't be conditional. If developers *really* want to remove this, they can explicitly pass `-Xlinker -no_application_extension` to override, which will be more robust than checking whether there are any flags at all.

Reply:
These are global flags and are generally used to link something outside of swiftpm (system library). This is very similar to the system modules case so I think it makes sense to disable this automatically. I can't think of a case where this would cause problems in practice.

Reply:
> This is very similar to the system modules case
True... yeah, this might be OK for now.